mlflow/server/auth/__init__.py

```diff
@@ -55,6 +142,244 @@ def make_basic_auth_response() -> Response:
+    UpdateRun: validate_can_update_run,
+    LogMetric: validate_can_update_run,
+    LogBatch: validate_can_update_run,
+    LogModel: validate_can_update_run,
+    SetTag: validate_can_update_run,
+    DeleteTag: validate_can_update_run,
+    LogParam: validate_can_update_run,
+    GetMetricHistory: validate_can_read_run,
+    ListArtifacts: validate_can_read_run,
+    # Routes for model registry
+    GetRegisteredModel: validate_can_read_registered_model,
+    DeleteRegisteredModel: validate_can_delete_registered_model,
+    UpdateRegisteredModel: validate_can_update_registered_model,
+    RenameRegisteredModel: validate_can_update_registered_model,
+    GetLatestVersions: validate_can_read_registered_model,
+    CreateModelVersion: validate_can_update_registered_model,
+    GetModelVersion: validate_can_read_registered_model,
+    DeleteModelVersion: validate_can_delete_registered_model,
+    UpdateModelVersion: validate_can_update_registered_model,
+    TransitionModelVersionStage: validate_can_update_registered_model,
+    GetModelVersionDownloadUri: validate_can_read_registered_model,
+    SetRegisteredModelTag: validate_can_update_registered_model,
+    DeleteRegisteredModelTag: validate_can_update_registered_model,
+    SetModelVersionTag: validate_can_update_registered_model,
+    DeleteModelVersionTag: validate_can_delete_registered_model,
+    SetRegisteredModelAlias: validate_can_update_registered_model,
+    DeleteRegisteredModelAlias: validate_can_delete_registered_model,
+    GetModelVersionByAlias: validate_can_read_registered_model,
+}
+
+
+def get_before_request_handler(request_class):
+    return BEFORE_REQUEST_HANDLERS.get(request_class)
+
+
+BEFORE_REQUEST_VALIDATORS = {
+    (http_path, method): handler
+    for http_path, handler, methods in get_endpoints(get_before_request_handler)
+    for method in methods
+}
+
+BEFORE_REQUEST_VALIDATORS.update(
+    {
+        (ROUTES.GET_USER, "GET"): validate_can_read_user,
+        (ROUTES.UPDATE_USER_PASSWORD, "PATCH"): validate_can_update_user_password,
+        (ROUTES.UPDATE_USER_ADMIN, "PATCH"): validate_can_update_user_admin,
+        (ROUTES.DELETE_USER, "DELETE"): validate_can_delete_user,
+        (ROUTES.GET_EXPERIMENT_PERMISSION, "GET"): validate_can_manage_experiment,
+        (ROUTES.CREATE_EXPERIMENT_PERMISSION, "POST"): validate_can_manage_experiment,
+        (ROUTES.UPDATE_EXPERIMENT_PERMISSION, "PATCH"): validate_can_manage_experiment,
+        (ROUTES.DELETE_EXPERIMENT_PERMISSION, "DELETE"): validate_can_manage_experiment,
+        (ROUTES.GET_REGISTERED_MODEL_PERMISSION, "GET"): validate_can_manage_registered_model,
+        (ROUTES.CREATE_REGISTERED_MODEL_PERMISSION, "POST"): validate_can_manage_registered_model,
+        (ROUTES.UPDATE_REGISTERED_MODEL_PERMISSION, "PATCH"): validate_can_manage_registered_model,
+        (ROUTES.DELETE_REGISTERED_MODEL_PERMISSION, "DELETE"): validate_can_manage_registered_model,
+    }
+)
+
+
 def _before_request():
```

Comment:
```suggestion
@catch_mlflow_exception
def _before_request():
```

Does this function need to be decorated with `catch_mlflow_exception` to handle an error gracefully?

Reply:
For example, renaming an experiment on UI throws the following error:

```
2023/04/27 17:55:25 ERROR mlflow.server: Exception on /ajax-api/2.0/mlflow/experiments/get-by-name [GET]
Traceback (most recent call last):
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/flask/app.py", line 2528, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/flask/app.py", line 1821, in full_dispatch_request
    rv = self.preprocess_request()
  File "/home/haru/miniconda3/envs/mlflow-dev-env/lib/python3.8/site-packages/flask/app.py", line 2312, in preprocess_request
    rv = self.ensure_sync(before_func)()
  File "/home/haru/Desktop/repositories/mlflow/mlflow/server/auth/__init__.py", line 407, in _before_request
    if not validator():
  File "/home/haru/Desktop/repositories/mlflow/mlflow/server/auth/__init__.py", line 233, in validate_can_read_experiment_by_name
    return _get_permission_from_experiment_name().can_read
  File "/home/haru/Desktop/repositories/mlflow/mlflow/server/auth/__init__.py", line 198, in _get_permission_from_experiment_name
    raise MlflowException(
mlflow.exceptions.MlflowException: Could not find experiment with name fasdfa
[2023-04-27 17:55:29 +0900] [285947] [DEBUG] POST /ajax-api/2.0/mlflow/runs/search
```

Reply:
that's right, i added it